test/cli/files.js

```diff
@@ -127,6 +127,23 @@ describe('files', () => runOnAndOff((thing) => {
       })
   })
 
+  it('add with piped argument', () => {
+    // echo 'src/init-files/init-docs/readme' | jsipfs files add
+    return ipfs('files add', { piped: 'src/init-files/init-docs/readme' })
```

Comment:
No, using stdin and passing a argument should not work the same way. Passing a argument should use that argument as a path to a file. Using stdin to use whatever comes from stdin as the content to add, without any filename.

Compare it to how go-ipfs works, and maybe it'll be a bit easier to understand.